### DISCUSSION TITLE
Edit Selected Text in VIM

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#974] [Share]
[Open edit-in-vim in Script Kit](https://scriptkit.com/api/new?name=edit-in-vim&url=https://gist.githubusercontent.com/johnlindquist/5813612d9b023709e6555b4e6e67dc50/raw/c9700e4c1b5b15451d0d77acd6bb455870d020ad/edit-in-vim.ts")

```js
// Name: Edit in VIM

import "@johnlindquist/kit"

let text = await getSelectedText()
let vimTextPath = tmpPath("vim.txt")
await writeFile(vimTextPath, text)

await term(`vim ${vimTextPath}`)

let contents = await readFile(vimTextPath, "utf-8")
await setSelectedText(contents)
```

**Comment by alexn-s** [comment]
neat. got to try it out. thanks for sharing

**Reply by johnlindquist** [reply]
@alexn-s Unfortunately, the terminal (`await term()`) is the one big feature not supported on Windows yet

**Reply by alexn-s** [reply]
ahh ok. looking forward to it in the future though
